x-pack/plugins/maps/public/connected_components/mb_map/transform_request.ts

```diff
@@ -20,8 +20,26 @@ const FONTS = getHttp().basePath.prepend(FONTS_API_PATH);
 const GETTILE = getHttp().basePath.prepend(MVT_GETTILE_API_PATH);
 const GETGRIDTILE = getHttp().basePath.prepend(MVT_GETGRIDTILE_API_PATH);
 
-export function transformRequest(url: string, resourceType: string | undefined) {
-  if (resourceType === 'Glyphs' && url.startsWith(FONTS)) {
+/**
+ * This URL could be used from inside a Worker which may have a different base
+ * URL. This function takes a string that may be a path and converts it to an
+ * absolute URL.
+ */
+function prepareAbsoluteUrl(pathOrUrl: string): string {
+  pathOrUrl = pathOrUrl.trim();
+  if (pathOrUrl.startsWith('/')) {
+    return new URL(pathOrUrl, window.location.origin).toString();
+  }
+  return pathOrUrl;
+}
+
+/**
+ * @param pathOrUrl - Assumed to be a full URL or a path starting with "/"
+ * @param resourceType - Indicator of what type of resource is being requested
+ */
+export function transformRequest(pathOrUrl: string, resourceType: string | undefined) {
+  const url = prepareAbsoluteUrl(pathOrUrl);
```

Comment:
Not sure I follow, this needed for the fix. I just renamed the original `url` variable to something more descriptive since paths, not full URLs, were being passed in.

Reply:
How is it needed for the fix? Now that you are no longer looking at the whole path, why is `prepareAbsoluteUrl ` needed? The logic is just looking for part of the path. I don't think this part of the path is different.

Reply:
chatted offline. `transformRequest` returns URL,  which needs to modified to ensure its always absolute. Thanks for explaining.